tests/ui/pathbuf_init_then_push.rs

```diff
@@ -0,0 +1,15 @@
+#![warn(clippy::pathbuf_init_then_push)]
```

Comment:
Could you also add some tests for code snippets that shouldn't trigger the lint. Something like:

```rs
let mut x = PathBuf::new();
println!("{}", x.display());
x.push("Duck");
```

And also this test, which should trigger the lint:

```rs
let mut path_buf = PathBuf::from("foo").join("bar");
path_buf.push("buz");
```